quic/s2n-quic/src/provider/tls.rs

```diff
@@ -204,7 +204,7 @@ mod default_tls {
         use super::*;
         use crate::provider::tls;
 
-        impl tls::Provider for Server {
+        impl<L: ConfigLoader> tls::Provider for Server<L> {
```

Comment:
I think this will only work for s2n-tls. The [rustls Server](https://github.com/aws/s2n-quic/blob/e31715845bc3d93ffbeb614284147a20defbb0c3/quic/s2n-quic-rustls/src/server.rs#L11) don't have a generic loader type. I'm surprised it didn't fail the windows build, which uses rustls by default. But that's a separate issue 😄

Reply:
Ah, interesting. I'll see if I can get the cfgs to cooperate, though testing the permutations here feels iffy.